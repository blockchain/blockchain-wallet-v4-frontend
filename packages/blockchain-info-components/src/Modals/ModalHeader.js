import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Icon } from '../Icons'
import { Text } from '../Text'

const Wrapper = styled.div`
  position: relative;
  display: flex;
  justify-content: ${props => (props.center ? 'center' : 'space-between')};
  align-items: center;
  width: 100%;
  padding: ${props => (props.tray ? '60px 10%' : '20px 30px')};
  box-sizing: border-box;
  border-bottom: 1px solid ${props => props.theme['gray-1']};
  padding-left: ${props => props.paddingHorizontal} !important;
  padding-right: ${props => props.paddingHorizontal} !important;

  & > :first-child {
    margin-right: 10px;
  }
`
const Header = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
`
const HeaderIcon = styled(Icon)`
  margin-right: 10px;
`

const ModalHeader = props => {
  const { closeButton, onClose, icon, children, paddingHorizontal } = props

  return (
    <Wrapper
      paddingHorizontal={paddingHorizontal}
      closeButton={closeButton}
      {...props}
    >
      <Header>
        {icon && (
          <HeaderIcon name={icon} size='28px' weight={300} color='gray-5' />
        )}
        <Text
          size='20px'
          weight={300}
          color='gray-5'
          className='headerText'
          data-e2e='modalHeaderText'
          capitalize
        >
          {children}
        </Text>
      </Header>
      {closeButton && (
        <Icon
          name='close-bold'
          size='16px'
          weight={300}
          color='gray-5'
          data-e2e='modalCloseButton'
          cursor
          onClick={onClose}
        />
      )}
    </Wrapper>
  )
}

ModalHeader.propTypes = {
  closeButton: PropTypes.bool,
  onClose: PropTypes.func,
  icon: PropTypes.string
}

ModalHeader.defaultProps = {
  closeButton: true
}

export default ModalHeader
