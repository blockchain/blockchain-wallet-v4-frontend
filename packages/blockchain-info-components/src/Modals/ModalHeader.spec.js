import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ModalHeader from './ModalHeader'

describe('ModalHeader component', () => {
  it('default renders correctly', () => {
    const component = shallow(<ModalHeader closeButton onClose={() => {}} icon='x' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
