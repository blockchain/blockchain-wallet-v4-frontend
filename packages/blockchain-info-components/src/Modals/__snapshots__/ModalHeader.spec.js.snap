// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader component default renders correctly 1`] = `
<ModalHeader__Wrapper
  closeButton={true}
  icon="x"
  onClose={[Function]}
>
  <ModalHeader__Header>
    <ModalHeader__HeaderIcon
      color="gray-5"
      name="x"
      size="28px"
      weight={300}
    />
    <Text
      altFont={false}
      capitalize={true}
      className="headerText"
      color="gray-5"
      cursor="inherit"
      data-e2e="modalHeaderText"
      italic={false}
      opacity={1}
      size="20px"
      uppercase={false}
      weight={300}
    />
  </ModalHeader__Header>
  <Icon
    color="gray-5"
    cursor={true}
    data-e2e="modalCloseButton"
    name="close-bold"
    onClick={[Function]}
    size="16px"
    weight={300}
  />
</ModalHeader__Wrapper>
`;
